src/layouts/product-integration-layout/components/header/style.module.css

```diff
@@ -1,46 +1,105 @@
-.header {
-	padding: 0;
-	background: var(--token-color-surface-faint);
-	max-width: 100%;
+.headerWrap {
+	@media (--dev-dot-tablet-up) {
+		display: grid;
+		grid-template-columns: auto 1fr;
+		grid-template-areas:
+			'icon content'
+			'icon badges';
+	}
+}
 
-	& .upperCard {
-		padding: 16px;
+.upper {
+	margin-bottom: 16px;
+	grid-area: content;
+
+	@media (--dev-dot-tablet-up) {
 		display: flex;
+		flex-direction: row-reverse;
 		justify-content: space-between;
-		@media (max-width: 600px) {
-			flex-direction: column;
-		}
-
-		& .left {
-			& h1 {
-				font-size: 38px;
-				line-height: 38px;
-				margin-top: 0;
-				margin-bottom: 4px;
-			}
-
-			& span {
-				font-size: 17px;
-			}
-		}
-
-		& .right {
-			& .versionDropdown {
-				@media (max-width: 600px) {
-					margin-top: 12px;
-				}
-			}
-		}
-	}
-
-	& .lowerCard {
-		background: var(--white);
-		border: 1px solid var(--wpl-neutral-25);
-		box-shadow: var(--token-elevation-mid-box-shadow);
-		border-radius: 5px;
-		padding: 12px;
+	}
+}
+
+.lower {
+	grid-area: badges;
+
+	@media (--dev-dot-tablet-up) {
 		display: flex;
 		justify-content: space-between;
 		align-items: center;
 	}
 }
+
+.iconTile {
+	grid-area: icon;
+	display: none;
+	margin-right: 12px;
+
+	@media (--dev-dot-tablet-up) {
+		display: block;
+	}
+}
+
+.titleDescriptionWrap {
+	margin-right: 12px;
+}
+
+.titleWrap {
+	margin: 12px 0 2px 0;
+	display: flex;
+	flex-wrap: wrap;
+	align-items: center;
+
+	@media (--dev-dot-tablet-up) {
+		margin-top: 0;
+		flex-wrap: unset;
+	}
+}
+
+.titleName {
+	color: var(--token-color-foreground-strong);
+	margin-right: 4px;
+}
+
+.titleOrganization {
+	color: var(--token-color-foreground-faint);
+}
+
+.description {
+	color: var(--token-color-foreground-primary);
+}
+
+.versionSelector {
+	width: 100%;
+
+	/** 
+	* style overrides for one-off mobile behavior
+	* TODO - consider supporting this within `DropdownDisclosure`
+	**/
+	& button {
```

Comment:
This isn't ideal but I think opening up `DropdownDisclosure` to support this style should be a separate effort.